Collection.c

```diff
@@ -89,22 +112,16 @@ unsigned conditionalMemCpy(void* pointer1, void* pointer2, unsigned elementCount
 	return count;
 }
 
-Collection* collection_collect(Collection* this, Collection* dst, void (*function)(void*)) {
-	void* memPointer = (char*)malloc(this->size);
-	memcpy(memPointer, this->list, this->size);
-	collection_init(dst, memPointer, this->size, this->typeSize);
-	collection_iterate(dst, function);
-	free(memPointer);
-	return dst;
+void collection_clone(Collection* this, Collection* dst){
+	this->list = dst->list;
```

Comment:
antes de asignar, usa el collection_free para liberar la memoria de this y despues si asignale a this list el dst list